lib/tasks/eu_exit_business_finder.rake

```diff
@@ -0,0 +1,78 @@
+    update_facet_values(subscriber_lists)
+  end
+
+  # Iterate through the affected subscriber lists and if we encounter any
+  # facet values that need replacing with a newer one, we insert the new one
+  # to the array (inserted 'in place' to retain original order). Then remove
+  # the old facet values in bulk and write only the unique values that remain.
+  def update_facet_values(subscriber_lists)
+    subscriber_lists.each do |list|
+      links = list.links[:facet_values][:any]
+      updated_links = links.dup # copy the links to avoid corrupting the loop
+      links.each_with_index do |value, index|
+        replacement = EuExitFacetMigrationConfig::facet_values_to_replace.find { |replace_val| replace_val[:old_facet_value] == value }
+        updated_links.insert(index, replacement[:new_facet_value]) if replacement
+      end
+      updated_links = updated_links - EuExitFacetMigrationConfig::facet_values_to_remove
+      updated_links = updated_links.uniq
+      list.write_attribute('links', facet_values: { any: updated_links })
+      list.save
+    end
+  end
+end
+
+module EuExitFacetMigrationConfig
+  def self.facet_values_to_replace
+    [
+      {
+        "old_facet_value": "9f3476e1-8ff0-455d-a14e-003236b2797c",
+        "new_facet_value": "53f9ce4c-7cbb-447f-bdf1-a9b022896d3a"
+      },
+      {
+        "old_facet_value": "b4e507df-f067-4749-9468-3de120775216",
+        "new_facet_value": "24fd50fa-6619-46ca-96cd-8ce90fa076ce"
+      },
+      {
+        "old_facet_value": "356f46a0-17d3-4ba4-8952-8c02244f9045",
+        "new_facet_value": "18c3892e-8a0e-4884-906e-5938380eceee"
+      },
+      # "34a6edd0-46ea-4a76-ae80-8c96709d4f5" is worth noting as
+      # it's the result of a merge between two facets. The
+      # find-and-replace works the same way regardless.
+      {
+        "old_facet_value": "1e3e8abd-135d-4844-afa8-5c51df3d3c57",
+        "new_facet_value": "34a6edd0-46ea-4a76-ae80-8c96709d4f59"
+      },
+      {
+        "old_facet_value": "c1d8057c-76bf-431c-9ac8-6281a4b7b9ca",
+        "new_facet_value": "34a6edd0-46ea-4a76-ae80-8c96709d4f59"
+      },
+      # "94b3cfe2-af89-4744-b8d7-7fc79edcbc85" is a special case
+      # as it's been split into two facets. We should assume the
+      # user wants to remain subscribed to both.
+      {
+        "old_facet_value": "94b3cfe2-af89-4744-b8d7-7fc79edcbc85",
+        "new_facet_value": "9d54c591-f5ca-4d0c-a484-12d5591987cb"
+      },
+      {
+        "old_facet_value": "94b3cfe2-af89-4744-b8d7-7fc79edcbc85",
+        "new_facet_value": "afd45eef-743a-417d-9245-3eab8322116d"
+      }
+    ]
+  end
+
+  def self.facet_values_to_remove
+    [
+      "7536c0c4-fb41-43f4-a2c4-08f4fa9f5427",
+      "5faa1741-fc55-4110-b342-de92f6324118",
+      "14cf2a68-3297-44d3-ba01-a4426845b1b8",
+      "040649fc-4e2c-4028-b846-77fe3eebd1f7",
+      "94b3cfe2-af89-4744-b8d7-7fc79edcbc85"
+    ] + facet_values_to_replace.map { |f| f[:old_facet_value] }.uniq
```

Comment:
If you use the `map` method that I've suggested on line 17, you probably won't need to add the `old_facet_value` values here, since they'll get replaced when doing the `map`.